src/services/query.ts

```diff
@@ -30,10 +30,23 @@ const schemaConstructor = Joi.object({
   comunica: Joi.object().required(),
 });
 
-export interface QueryResult {
-  distribution: Distribution;
-  terms: Term[];
+export type QueryResult = Success | TimeoutError | ServerError;
```

Comment:
These type and class names make perfect sense, but also are somewhat difficult to distinguish at first sight. For example: what's the difference between a `QueryResult` and just a `Result`? And why is the query error named `QueryError` but the query success just `Success`?

Reply:
Good point! The reason is a bad one: I already added a `Result`, so this had to be named differently and became `QueryResult`. I simplified the type names; are they clearer now?

Reply:
Perfect, thanks!